Modules/Scripted/DMRIInstall/DMRIInstall.py

```diff
@@ -58,20 +85,43 @@ def setup(self):
     ScriptedLoadableModuleWidget.setup(self)
 
     self.textBox = ctk.ctkFittedTextBrowser()
+    self.textBox.setOpenExternalLinks(True) # Open links in default browser
     self.textBox.setHtml(DMRIInstall.helpText)
     self.parent.layout().addWidget(self.textBox)
 
     #
     # Apply Button
     #
-    self.applyButton = qt.QPushButton("Open Extension Manager")
-    self.applyButton.toolTip = 'Install the "SlicerDMRI" extension from the Diffusion category.'
+    self.applyButton = qt.QPushButton("Install SlicerDMRI")
+    self.applyButton.toolTip = 'Installs the "SlicerDMRI" extension from the Diffusion category.'
     self.applyButton.icon = qt.QIcon(":/Icons/ExtensionDefaultIcon.png")
     self.applyButton.enabled = True
     self.applyButton.connect('clicked()', self.onApply)
     self.parent.layout().addWidget(self.applyButton)
 
     self.parent.layout().addStretch(1)
 
+
+  def onError(self):
+    self.applyButton.enabled = False
+    self.textBox.setHtml(DMRIInstall.errorText)
+    return
+
+
   def onApply(self):
-    slicer.app.openExtensionsManagerDialog()
+    emm = slicer.app.extensionsManagerModel()
+
+    if emm.isExtensionInstalled("SlicerDMRI"):
+      self.textBox.setHtml("<h4>SlicerDMRI is already installed.<h4>")
+      self.applyButton.enabled = False
+      return
+
+    md = emm.retrieveExtensionMetadataByName("SlicerDMRIFoo")
```

Comment:
Is `SlicerDMRIFoo` the expected name ?

Reply:
Hah, oops. I forgot to tack out the failure-mode test. Thanks!